packages/protect/__tests__/nested-models.test.ts

```diff
@@ -0,0 +1,422 @@
+import 'dotenv/config'
+import { describe, expect, it, vi } from 'vitest'
+
+import { LockContext, protect, csTable, csColumn, csValue } from '../src'
+
+const users = csTable('users', {
+  email: csColumn('email').freeTextSearch().equality().orderAndRange(),
+  address: csColumn('address').freeTextSearch(),
+  name: csColumn('name').freeTextSearch(),
+  example: {
+    field: csValue('example.field'),
+    nested: {
+      deeper: csValue('example.nested.deeper'),
+    },
+  },
+})
+
+type User = {
+  id: string
+  email?: string | null
+  createdAt?: Date
+  updatedAt?: Date
+  address?: string | null
+  notEncrypted?: string | null
+  example: {
+    field: string | undefined | null
+    nested?: {
+      deeper: string | undefined | null
+    }
+  }
+}
+
```

Comment:
I think it'd be worth adding some tests for:
- a mix of plaintext and encrypted nested fields
- partial matches of paths